src/unix/bsd/freebsdlike/freebsd/mod.rs

```diff
@@ -990,6 +996,21 @@ s! {
         pub _flags: u32,
         pub _clockid: u32,
     }
+
+    // When sigevent was first added to libc, Rust still didn't support unions.
+    // So the definition only included one of the union's member.  This
+    // structure exists for backwards-compatibility with consumers that still
+    // try to access that one member.
+    #[doc(hidden)]
+    pub struct sigevent_0_2_126 {
+        pub sigev_notify: ::c_int,
+        pub sigev_signo: ::c_int,
+        pub sigev_value: ::sigval,
+        pub sigev_notify_thread_id: ::lwpid_t,
```

Comment:
I would prefer if this was a `union` that just contains a single field. This forces the access to the `sigev_notify_thread_id` field to be in an `unsafe` block, which is necessary since this may be an uninitialized field in the union.

Reply:
But the whole point of this structure is for backwards-compatibility with older libc versions, and they didn't require `unsafe`.  Wouldn't it break backwards-compatibility then, if we put this field into a union?

Reply:
It just needs to be backwards-compatible enough for existing users, which is basically just nix. Does nix access this field outside of an unsafe block?

Reply:
Yes, existing Nix does not use `unsafe`.

Reply:
Also, I don't think that it should be considered unsafe to access the union member when using the backwards-compatible definition.  Since it's defined as a struct with a bunch of padding fields, the whole structure should always be initialized, and it shouldn't be a problem to read from the "union" member.

Reply:
I think this is most likely fine, but I would like a second opinion from @JohnTitor.

Reply:
Hmm, I'm not sure if I've got 100% of the context here so don't take my thought as a strong opinion, but I think it doesn't have to be _perfectly correct_ while it's just for back-compat.